src/plugins.c

```diff
@@ -2000,7 +2000,7 @@ gboolean geany_plugin_register_proxy(GeanyPlugin *plugin, const gchar **extensio
 	foreach_list(node, active_proxies.head)
 	{
 		proxy = node->data;
-		g_return_if_fail(p != proxy->plugin);
+		g_return_val_if_fail(p != proxy->plugin, FALSE);
```

Comment:
@kugel- based on the comment above the loop, why does this need to print an error if the plugin is already registered?

The `g_return_*` functions are only meant for programming errors, see [here](https://developer.gnome.org/glib/stable/glib-Warnings-and-Assertions.html#glib-Warnings-and-Assertions.description)